scrape/scrape.go

```diff
@@ -662,7 +662,11 @@ mainLoop:
 				time.Since(last).Seconds(),
 			)
 		}
-		b := sl.buffers.Get(sl.lastScrapeSize)
+		b, ok := sl.buffers.Get(sl.lastScrapeSize).([]byte)
+		if !ok {
+			b = make([]byte, 0, sl.lastScrapeSize)
+			level.Error(sl.l).Log("msg", "buffer pool type assertion error")
+		}
```

Comment:
Mh, I think a failing assertion here is a hard programming bug just as much as passing a non-slice to `Put`. Maybe just cast it without the `ok` check and let it fail otherwise?